iam.tf

```diff
@@ -71,14 +71,14 @@ resource "databricks_group_member" "this" {
 
 resource "databricks_entitlements" "this" {
   for_each = {
-    for group, params in var.iam : group => params.entitlements
+    for group, params in var.iam : group => params
     if params.entitlements != null
   }
 
   group_id                   = databricks_group.this[each.key].id
-  allow_cluster_create       = contains(each.value, "allow_cluster_create")
-  allow_instance_pool_create = contains(each.value, "allow_instance_pool_create")
-  databricks_sql_access      = contains(each.value, "databricks_sql_access")
+  allow_cluster_create       = contains(each.value.entitlements, "allow_cluster_create")
```

Comment:
to conditionally filter out null values or even presence of 'entitlement' parameter, please try these equation for all 'contains' parameters in this resource:
```hcl
allow_cluster_create = contains(lookup(each.value, "entitlements", []), "allow_cluster_create")
```

Reply:
Modified. Couldn't find any difference in behevior